numpyro/distributions/distribution.py

```diff
@@ -217,23 +216,16 @@ def rvs(self, *args, **kwargs):
         size = kwargs.pop('size', args.pop() if len(args) > self.numargs else None)
 
         args, _, _ = self._parse_args(*args, **kwargs)
-        # XXX we might not need to verify that args is empty for multivariate distributions
-        args = _promote_args("rvs", *args)
 
-        # TODO: make this code compatible to mvn distribution
         if not size:
-            size = args[-1].shape[:-1]
+            size = self._batch_shape(*args)
```

Comment:
I am curious to know what is the functionality of this class? If it is just this behavior of adjusting size to account for the event shape, can we not do this in `jax_discrete` and `jax_continuous` now that you have already defined `event_shape` and `batch_shape`.

Reply:
This class mainly provides a `parse_args` method different from `rv_discrete, rv_continuous` which includes `loc` and `scale`. On the other hand, the `size` pattern is also different from the other two classes.
For continuous dist, if size is not provided, then `size` is the broadcasted shape of args + loc + scale.
For discrete dist, if size is not provided, then `size` is the broadcasted shape of args + loc. For multivariate distribution, this pattern is not true any more:
+ for mvn: `size` = broadcasted(mean.shape, cov.shape[:-1])
+ for dirichlet: `size = alpha.shape[:-1]`
+ for multinomial: `size = p.shape[:-1]`, n is int
+ ...
> can we not do this in jax_discrete and jax_continuous now
Yes, thanks for suggestion! I believe we can make a common `rvs` method for `jax_generic` based on `_batch_shape` method. I'll make a follow-up PR for it.

Reply:
Thanks for explaining. 

>  I'll make a follow-up PR for it if you also think so.

I was asking more for my clarification; I actually think we shouldn't spend more time on this than is needed to unblock any work that you are doing, because I am not convinced that hacks like `parse_args` need to be built up on. :)